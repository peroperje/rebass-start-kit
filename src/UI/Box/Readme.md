Wrapper for [rebass Box Component](https://rebassjs.org/box)

```js
<Box bg="blue" color="white">
  Box content
</Box>
```
