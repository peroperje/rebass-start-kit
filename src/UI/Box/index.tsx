import React from 'react';
import * as rebass from 'rebass';

export type BoxProps = rebass.BoxProps;

const BoxR = rebass.Box;

const Box: React.FC<BoxProps> = (props: BoxProps) => <BoxR {...props} />;

export default Box;
