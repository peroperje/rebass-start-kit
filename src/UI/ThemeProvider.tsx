import React from "react";
import { ThemeProvider as EmotionThemeProvider } from "emotion-theming";

import theme from "./theme";

export interface Props {
  children: React.ReactNode;
}

const ThemeProvider: React.FC<Props> = ({ children }: Props): JSX.Element => (
  <EmotionThemeProvider theme={theme}>{children}</EmotionThemeProvider>
);

export default ThemeProvider;
