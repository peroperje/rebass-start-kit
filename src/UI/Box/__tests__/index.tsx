import React from "react";
import render from "react-test-renderer";

import ThemeProvider from "../../ThemeProvider";

import Box from "../index";

describe("Box", () => {
  it("Default snapshot", () => {
    const tree = render.create(
      <ThemeProvider>
        <Box>Box content</Box>
      </ThemeProvider>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
