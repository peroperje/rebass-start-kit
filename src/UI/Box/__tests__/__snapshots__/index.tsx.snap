// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Default snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

<div
  className="emotion-0"
>
  Box content
</div>
`;
